File: pdfGen2Java/app/src/main/java/com/pmgohil/pdfgen2java/MainActivity.java
package com.pmgohil.pdfgen2java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.pdf.PdfDocument;
import android.os.Bundle;
import android.os.Environment;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

import java.io.File;
import java.io.FileOutputStream;
import java.io.FileWriter;
import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    Button btnCreate;
    EditText editText;
    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnCreate = (Button) findViewById(R.id.create);
        editText = (EditText) findViewById(R.id.edittext);
        tv = (TextView) findViewById(R.id.tv);
        btnCreate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Context con = getApplicationContext();
                String fnm = "p.txt";
                String body = "I'm Prakash Gohil.";
                writeFileOnInternalStorage(con, fnm, body);

                //createPdf(editText.getText().toString());
            }
        });
    }

    public void writeFileOnInternalStorage(Context mcoContext, String sFileName, String sBody) {
        File dir = new File(mcoContext.getFilesDir(), "mydir");
        if (!dir.exists()) {
            dir.mkdir();
        }

        try {
            File gpxfile = new File(dir, sFileName);
            FileWriter writer = new FileWriter(gpxfile);
            writer.append(sBody);
            writer.flush();
            writer.close();
            tv.setText("Done");
        } catch (Exception e) {
            e.printStackTrace();
            tv.setText(e.toString());
        }
    }

    private void createPdf(String sometext) {
        // create a new document
        PdfDocument document = new PdfDocument();
        // crate a page description
        PdfDocument.PageInfo pageInfo = new PdfDocument.PageInfo.Builder(300, 600, 1).create();
        // start a page
        PdfDocument.Page page = document.startPage(pageInfo);
        Canvas canvas = page.getCanvas();
        Paint paint = new Paint();
        paint.setColor(Color.RED);
        canvas.drawCircle(50, 50, 30, paint);
        paint.setColor(Color.BLACK);
        canvas.drawText(sometext, 80, 50, paint);
        //canvas.drawt
        // finish the page
        document.finishPage(page);
// draw text on the graphics object of the page
        // Create Page 2
        pageInfo = new PdfDocument.PageInfo.Builder(300, 600, 2).create();
        page = document.startPage(pageInfo);
        canvas = page.getCanvas();
        paint = new Paint();
        paint.setColor(Color.BLUE);
        canvas.drawCircle(100, 100, 100, paint);
        document.finishPage(page);
        // write the document content
        String directory_path = Environment.getExternalStorageDirectory().getPath() + "/mypdf/";
        File file = new File(directory_path);
        if (!file.exists()) {
            file.mkdirs();
        }
        String targetPdf = directory_path + "test-2.pdf";
        File filePath = new File(targetPdf);
        try {
            document.writeTo(new FileOutputStream(filePath));
            Toast.makeText(this, "Done", Toast.LENGTH_LONG).show();
        } catch (IOException e) {
            Log.e("prakash", "error " + e.toString());
            Toast.makeText(this, "Something wrong: " + e.toString(), Toast.LENGTH_LONG).show();
            tv.setText(e.toString());
        }
        // close the document
        document.close();
    }
}

File: Toast/app/src/main/java/com/example/toast/MainActivity.kt
package com.example.toast

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private var toast: Toast? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn1 = findViewById<Button>(R.id.btn1)
        btn1.setOnClickListener() {
            cancel()
            toast = Toast.makeText(this@MainActivity, "10", Toast.LENGTH_SHORT)
            toast!!.show()
        }

        val btn2 = findViewById<Button>(R.id.btn2)
        btn2.setOnClickListener() {
            cancel()
            toast = Toast.makeText(this@MainActivity, "100", Toast.LENGTH_SHORT)
            toast!!.show()
        }

        val btn3 = findViewById<Button>(R.id.btn3)
        btn3.setOnClickListener() {
            cancel()
            toast = Toast.makeText(this@MainActivity, "1000", Toast.LENGTH_SHORT)
            toast!!.show()
        }

        findViewById<Button>(R.id.btn4).setOnClickListener() {
            cancel()
            toast = Toast.makeText(this@MainActivity, "Prakash", Toast.LENGTH_SHORT)
            toast?.show()
        }
    }

    private fun cancel() {
        if (toast != null) {
            toast?.cancel()
        }
    }
}